src/utils/self_test/self_test.c

```diff
@@ -1674,119 +1918,192 @@ int main(int argc, char *argv[])
 	int				 c;
 	int				 j;
 	int				 ret = 0;
-	struct st_endpoint		*endpts = NULL;
-	struct st_endpoint		*ms_endpts = NULL;
-	uint32_t			 num_endpts = 0;
-	uint32_t			 num_ms_endpts = 0;
-	int				 output_megabits = 0;
-	int16_t				 buf_alignment =
-		CRT_ST_BUF_ALIGN_DEFAULT;
-	char				*attach_info_path = NULL;
+	int				 dump = 0;
+
+	static struct option long_options[] = {
+		{"file-name", required_argument, 0, 'f'},
+		{"config", required_argument, 0, 'c'},
+		{"display", no_argument, 0, 'd'},
+
+		{"group-name", required_argument, 0, 'g'},
+		{"master-endpoint", required_argument, 0, 'm'},
+		{"endpoint", required_argument, 0, 'e'},
+		{"message-sizes", required_argument, 0, 's'},
+		{"repetitions-per-size", required_argument, 0, 'r'},
+		{"max-inflight-rpcs", required_argument, 0, 'i'},
+		{"align", required_argument, 0, 'a'},
+		{"Mbits", no_argument, 0, 'b'},
+		{"singleton", no_argument, 0, 't'},
+		{"randomize-endpoints", no_argument, 0, 'q'},
+		{"path", required_argument, 0, 'p'},
+		{"nopmix", no_argument, 0, 'n'},
+		{"help", no_argument, 0, 'h'},
+		{0, 0, 0, 0}
+	};
+	extern int			optind;
```

Comment:
(style)  externs should be avoided in .c files

Reply:
optind not defined in include file.  Need here.